### PR TITLE
Jobrouter initial commit

### PR DESCRIPTION
**Pull request by cparisineti**
# Description

PR for Internal review of Job Router's Java sdk by JobRouter team.

### CODE REVIEW

**Comment by rcoleobrien** [comment]
I know Raj was having issues generating some of our classes that have no properties e.g DeclineJobOfferResponse for other SDK's and had to manually declare them. I think we'll also need to do this for now. I'm checking with swashbuckle team to see if we can get around this

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/CommunicationBearerTokenCredential.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * This class serves as a CommunicationTokenCredential wrapper that
+ * allows using BearerAuthenticationPolicy in different clients
+ */
+public class CommunicationBearerTokenCredential implements TokenCredential {
```
Comment (williamzhao87) [inline sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/CommunicationBearerTokenCredential.java:17]: Do we need this class? I believe we're only using HMAC auth.

**Review by williamzhao87 — CHANGES_REQUESTED** [review]

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAsyncClient.java [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+ *
+ * </pre>
+ * <!-- end com.azure.communication.chat.chatasyncclient.instantiation -->
+ *
+ * <p>View {@link JobRouterClientBuilder this} for additional ways to construct the client.</p>
+ *
+ * @see JobRouterClientBuilder
+ */
+@ServiceClient(builder = JobRouterClientBuilder.class, isAsync = true)
+public class JobRouterAsyncClient {
```
[the remark this answers is not in the document]
Comment (cparisineti) [reply sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAsyncClient.java:43]: Yeah. I will rename it to RouterClient.

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterClient.java [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+    }
+
+    // Offers
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<AcceptJobOfferResponse> acceptJobAction(String offerId, String workerId) {
+        return this.acceptJobActionWithResponse(offerId, workerId, null);
+    }
+
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Response<AcceptJobOfferResponse> acceptJobActionWithResponse(String offerId, String workerId, Context context) {
```
[the remark this answers is not in the document]
Comment (cparisineti) [reply sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterClient.java:280]: Makes sense.

File: sdk/communication/azure-communication-jobrouter/src/main/resources/azure-communication-chat.properties [hunk 4]
```diff
@@ -0,0 +1,2 @@
+name=${project.artifactId}
```
[the remark this answers is not in the document]
Comment (cparisineti) [reply sdk/communication/azure-communication-jobrouter/src/main/resources/azure-communication-chat.properties:1]: woops.

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/CommunicationBearerTokenCredential.java (hunk 1 above)
Comment (cparisineti) [reply sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/implementation/CommunicationBearerTokenCredential.java:17]: No, will remove.

**Comment by cparisineti** [comment]
William's comments are addressed in this PR https://github.com/Azure/azure-sdk-for-java/pull/29200. Moved to another branch as pipeline was not running on cparisineti:jobrouter-initial-commit branch.

**Pull request closed without merging**